File: C/Algorithms/559.c
// Definition for a Node.
struct Node {
  int val;
  int numChildren;
  struct Node **children;
};

int depth(struct Node *root) {
  // base case.
  if (root->numChildren == 0)
    return 1;
  // recurse.
  int max = 0;
  for (int i = 0; i < root->numChildren; i++) {
    int value = depth(root->children[i]);
    if (value > max) {
      max = value;
    }
  }
  return max + 1;
}

int maxDepth(struct Node *root) {
  if (!root) {
    return 0;
  }
  return depth(root);
}

File: Javascript/Algorithms/559.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node} root
 * @return {number}
 */
let maxDepth = root => {
    let depth = node => {
        if (node.children.length == 0){
            return 1;
        }
        return 1 + _.max(node.children.map((n) => depth(n)));
    }
    return root ? depth(root) : 0;
};
